Install root dependencies

theworkflow opened the pull request:
Install root dependencies when `package.json` is present. Then moves them to the `demeteorized` bundle.

Closes #181

theworkflow committed:
Add functionality to install root package.json dependencies

theworkflow committed:
Copy node_modules to demeteorized bundle

theworkflow committed:
Fix FS mode when accessing package.json

fiveisprime commented on lib/move-dependencies.js:
```diff
@@ -0,0 +1,16 @@
+    if (err && err.code === 'ENOENT') return done();
+    if (err) return done(err);
+
+    FS.copy(source, destination, done);
```
Should this be called `copy-dependencies.js` instead? 🙃

fiveisprime commented on package.json:
```diff
@@ -24,8 +24,9 @@
-    "hoek": "4.x.x",
-    "cross-spawn": "3.0.1"
+    "cross-spawn": "3.0.1",
+    "fs-extra": "^0.30.0",
```
Remove the `^`

theworkflow committed:
Rename moveDependencies to copyDependencies and specify version
of `fs-extra` to install

Commit:
Updated to pass standard lint.

Remove unnecessary semicolons so the build passes. I broke it so I figured
I would fix it. <3

fiveisprime commented:
I fixed the build issues and resolved the conflict with `master` since I broke that..

theworkflow commented:
Merge conflicts have been resolved.

jackboberg commented:
@theworkflow Are you considering closing #243 with this PR?

theworkflow commented:
@jackboberg Not initially. I had the thought process that each function/file is an individual task. Updating the `engines` block of the `package.json` would be a different task. If you feel it should be handled best in this PR I'm down for adding it.

jackboberg reviewed (approved):
A separate PR seems fine, I think it could be a part of [`find-node-version`](https://github.com/onmodulus/demeteorizer/blob/master/lib/find-node-version.js).
